docs/src/architecture/state.md

```diff
@@ -11,89 +11,60 @@ the system maintains 3 databases to describe the state:
 
-However, we need to explicitly store only the unconsumed public notes and enough info to construct membership proofs against them. Private notes, as well as public notes which have already been consumed, can be safely discarded. Such notes would still remain in the accumulator, but there is no need to store them explicitly as the append-only accumulator can be updated without knowing all items stored in it. This would reduce actual storage requirements to a fraction of the database's nominal size.
-
-Moreover, since notes are not meant as long-lived objects, we can impose some lifetimes restrictions on them. For example, we could say that the system will store only $2^{35}$ most recent notes (a note not consumed in this timeframe becomes un-spendable). This can be easily done by discarding old commitment roots once the number of roots exceeds 35. At 1K tps this would mean that notes would be discarded after about 1 year, but this would also mean that the size of the note database will never grow beyond about 1TB.
+> (Is this the case in v1?) Moreover, since notes are not meant as long-lived objects, we can impose some lifetimes restrictions on them. For example, we could say that the system will store only $2^{35}$ most recent notes (a note not consumed in this timeframe becomes un-spendable). This can be easily done by discarding old commitment roots once the number of roots exceeds 35. At 1K tps this would mean that notes would be discarded after about 1 year, but this would also mean that the size of the note database will never grow beyond about 1TB.
 
 ### Nullifier database
 
-With nullifier database we want to achieve the following properties:
-1. We need to be able to check that a given nullifier is not in the database. This is needed to ensure that notes consumed in a transaction haven't been already consumed.
-2. We need to be able to prove that a given nullifier is not in the database. This is needed for state transition proofs we want to submit to L1.
-3. We need to be able to add new nullifiers to the database. This would be done by an operator at the time when they create a new block.
-
-To satisfy these properties we can use a Sparse Merkle tree which maps nullifiers to block heights at which they were created. For example, in the diagram below, the tree contains 2 nullifiers: nullifier `01` was inserted into the database at block height $4$, while nullifier `10` was inserted into the database at block height $5$.
+Nullifiers provide information whether a specific note has been consumed yet. Nullifiers are stored in a Sparse Merkle Tree which maps nullifiers to block heights at which they were created.  Using this, one can check and even prove that a given nullifier is not in the database. There will be one tree per epoch (~3 month) and Miden nodes store always at least the last two trees. However, the roots of the old trees are still stored.
```

Comment:
have we concluded on an epoch being 3 months?

Reply:
I think 3 moths is a good placeholder - but 6 moths or 1 year could also be reasonable alternatives - so, we should discuss this.

Reply:
Let's discuss in our sync meeting today. And I will add a discussion in this repo.